src/Implementation/System/SymbolicDescription.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Numerics;
+using Symbolica.Abstraction;
+using Symbolica.Expression;
+
+namespace Symbolica.Implementation.System
+{
+    internal sealed class SymbolicDescription : IPersistentDescription
+    {
+        private readonly ulong _size;
+        private readonly long _offset;
+
+        private SymbolicDescription(ulong size, long offset)
+        {
+            _size = size;
+            _offset = offset;
+        }
+
+        public (long, IPersistentDescription) Seek(long offset, uint whence)
+        {
+            var result = whence switch
+            {
+                0U => offset,
+                1U => _offset + offset,
+                2U => (long)_size + offset,
+                _ => -1L
+            };
+
+            return (result, new SymbolicDescription(_size, result));
+        }
+
+        public int Read(ISpace space, IMemory memory, IExpression address, int count)
+        {
+            if (_offset < 0)
```

Comment:
I'm not sure if this is correct, but from my quick skimming of the documentation it looks like if the offset is negative then it means something's gone wrong in a previous file operation. So I introduced a new `StateError` here, which may or may not be the correct way to handle this, assuming it is even an error that should be handled here.